opennms-webapp/src/main/webapp/heatmap/heatmap-box.jsp

```diff
@@ -115,136 +115,144 @@
-                ],
-                "colorStops": [
-                    {"val": 1.0, "color": "#CC0000"},
-                    {"val": 0.4, "color": "#FF3300"},
-                    {"val": 0.2, "color": "#FF9900"},
-                    {"val": 0.1, "color": "#FFCC00"},
-                    {"val": 0.0, "color": "#336600"}
-                ],
-                "labelsEnabled": true,
-                "nodeData": {
-                    "id": "<%=heatmap%>",
-                    "children": children
+                  if ("nodesByCategory".equals(heatmap) || "nodesByForeignSource".equals(heatmap) || "nodesByMonitoredService".equals(heatmap)) {
+                %>
+                location.href = "/opennms/element/node.jsp?node=" + nodes[0].elementId
+                <%
+                  }
+                %>
+            };
+
+            var url = "<%=url%>";
+            var children;
+
+            function refresh() {
+                var height = $(window).height() - 105 - $("#treemap").offset().top;
+
+                if (height < 0) {
+                    height = $(window).width();
                 }
-            }).bind('treemapclick', mouseclickHandler);
-        }
 
-        $( window ).resize(function() {
-            refresh();
-        });
+                height = Math.min(height, $(window).width());
 
-        $(document).ready(function () {
-            $.getJSON(url, function (data) {
-                children = data.children;
+                $("#treemap").treemap({
+                    "dimensions": [
+                        $("#treemap").width(),
+                        height
+                    ],
+                    "colorStops": [
+                        {"val": 1.0, "color": "#CC0000"},
+                        {"val": 0.4, "color": "#FF3300"},
+                        {"val": 0.2, "color": "#FF9900"},
+                        {"val": 0.1, "color": "#FFCC00"},
+                        {"val": 0.0, "color": "#336600"}
+                    ],
+                    "labelsEnabled": true,
+                    "nodeData": {
+                        "id": "<%=heatmap%>",
+                        "children": children
+                    }
+                }).bind('treemapclick', mouseclickHandler);
+            }
+
+            $(window).resize(function() {
                 refresh();
             });
+
+            $(document).ready(function () {
+                $.getJSON(url, function (data) {
+                    children = data.children;
+                    refresh();
+                });
+            });
         });
     </script>
     <div class="panel-footer">
-        <span class="glyphicon glyphicon-retweet" aria-hidden="true"></span>&nbsp;
-        <%
-            if ("outages".equals(mode)) {
-        %>
-        <a href="<%=request.getRequestURI()%>?mode=alarms&heatmap=<%=heatmap%>&category=<%=category%>&foreignSource=<%=foreignSource%>&monitoredService=<%=monitoredService%>">Alarms</a> / <b>Outages</b>
```

Comment:
I only notice these things because github adds red highlights to the &s in the PR UI. :)